modules/ui/combobox.js

```diff
@@ -390,10 +390,37 @@ export function uiCombobox(context, klass) {
                 })
                 .attr('title', function(d) { return d.title; })
                 .each(function(d) {
+                    var option = d3_select(this);
                     if (d.display) {
-                        d.display(d3_select(this));
+                        d.display(option);
                     } else {
-                        d3_select(this).text(d.value);
+                        option.text(d.value);
+                    }
+                    // Only for comment combobox: add cross button, floated right, minimal style
+                    if (klass === 'comment') {
+                        option.append('button')
+                            .attr('class', 'combobox-delete')
+                            .attr('title', 'Delete this comment')
+                            .style('float', 'right')
+                            .style('font-size', '16px')
+                            .style('cursor', 'pointer')
+                            .text('×')
+                            .on('mousedown', function(event) {
+                                event.stopPropagation();
+                                event.preventDefault();
+                            })
+                            .on('click', function(event, d) {
+                                event.stopPropagation();
+                                event.preventDefault();
+                                let deleted = JSON.parse(localStorage.getItem('deletedChangesetComments') || '[]');
+                                let valueNorm = d.value.trim().toLowerCase();
+                                if (!deleted.map(s => s.trim().toLowerCase()).includes(valueNorm)) {
+                                    deleted.push(d.value);
+                                    localStorage.setItem('deletedChangesetComments', JSON.stringify(deleted));
```

Comment:
How will this get reset? Like, when I click it 100 times, all those will be in the stack forever, right? Should we maybe allow for a max of 100 items and remove the last items? Like a max array size…

Reply:
It should be safe to remove any stored entries that do not match any changeset that is currently returned by the OSM API anymore.